modules/core/src/main/java/org/apache/synapse/core/axis2/TimeoutHandler.java

```diff
@@ -209,6 +210,8 @@ private void processCallbacks() {
                                                                          callback.getAxis2OutMsgCtx()) +
                                  ", " + getServiceLogMessage(callback.getSynapseOutMsgCtx())) ;
                     }
+                    org.apache.synapse.MessageContext synapseOutMsgCtx = callback.getSynapseOutMsgCtx();
```

Comment:
should not we check if callback is null here? 
In SynapseCallbackReceiver that logic is there.

Reply:
no, it is already checked above.

Reply:
OK, Agreed.